sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala

```diff
@@ -335,7 +513,10 @@ class AggHelper(
     // We need to merge the aggregated batches into 1 before calling post process,
     // if the aggregate code had to split on a retry
     if (aggregatedSeq.size > 1) {
-      val concatted = concatenateBatches(metrics, aggregatedSeq)
+      val concatted =
+        withResource(aggregatedSeq) { _ =>
```

Comment:
I am confused by this? Was this a bug? This change feels wrong to me.

concatenateBatches has the contract that it will either close everything in toConcat, or if there is a single item in the sequence it will just return it without closing anything.  By putting it within a `withResource` it looks like we are going to double close the data in aggregatedSeq.

Reply:
`SpillableColumnarBatch` has the nasty habit (?) of hiding double closes from us (https://github.com/NVIDIA/spark-rapids/blob/branch-24.12/sql-plugin/src/main/scala/com/nvidia/spark/rapids/SpillableColumnarBatch.scala#L137). I'd like to remove this behavior with my spillable changes.

Reply:
I think I was mislead by https://github.com/NVIDIA/spark-rapids/blob/5be4bd5cd11cba034f370d5fdba6e18230193e66/sql-plugin/src/main/scala/com/nvidia/spark/rapids/GpuAggregateExec.scala#L955 on main branch, and thought concatenateBatches will not close input batches. Will revert this part